app/scripts/lib/WeakRefObjectMap.ts

```diff
@@ -0,0 +1,74 @@
+type WeakRefCompatibleObject<T> = {
+  [P in keyof T]: T[P] extends object ? WeakRef<T[P]> : never;
+};
+
+export class WeakRefObjectMap {
```

Comment:
Nit: It would be great to have TSDoc descriptions for this class and all public methods.

I get that it's mostly a Map implementation, but the goals of using WeakRef here are worth explaining. The part where we deconstruct the network client to store each property independently is super weird as well. I almost asked why we were doing that, before remembering the fact that we only use hard references to the network client properties not the client object itself.

Reply:
done here: https://github.com/MetaMask/metamask-extension/pull/23525/commits/a3770013832099fc91e51e849f3ff399b556d4f8